pkgs/development/python-modules/vxi11/default.nix

```diff
@@ -17,10 +18,14 @@ buildPythonPackage rec {
     sha256 = "1xv7chp7rm0vrvbz6q57fpwhlgjz461h08q9zgmkcl2l0w96hmsn";
   };
 
-  nativeCheckInputs = [ nose ];
-  checkPhase = ''
-    nosetests
-  '';
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/python-ivi/python-vxi11/pull/47.patch";
```

Comment:
URLs like this aren’t stable. This should ideally be `fetchpatch2 { url = "https://github.com/python-ivi/python-vxi11/commit/<hash>.patch?full_index=1"; … }` for each commit, but if that’s too onerous then it might be best to vendor this PR’s patch as a file instead. (In that case, please do leave a link to the PR in a comment.)

Reply:
Should I include this commit as part of the patch?
https://github.com/loft-orbital/python-vxi11/commit/00722b1b8810ac38bfb47e8c49437055b600dfff
I it put within the original pr

Reply:
Yeah, that fixes warnings on Python 3.12.

Reply:
Well applying 2nd patch causes to fail:
```
error: builder for '/nix/store/qz3h57h2075vg9vrsbp19qf0m172x731-python3.12-python-vxi11-0.9.drv' failed with exit code 1;
       last 10 log lines:
       > Running phase: patchPhase
       > applying patch /nix/store/wckh73xcab2h9676i43ds58a5g9d17p0-a8ad324d645d6f7215f207f2cc2988dc49859698.patch
       > patching file tests/test_vxi11.py
       > applying patch /nix/store/hxdp0j638l5f5s68151l597w4zyi40yc-eabc4de210912242293f1417bed97619e14520e2.patch
       > patching file tests/test_vxi11.py
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 1 out of 1 hunk ignored -- saving rejects to file tests/test_vxi11.py.rej
       > patching file vxi11/vxi11.py
```
But only the first is technically needed

Reply:
That’s because you’re applying the merge commit that contains both the previous commits in the PR. You want `00722b1b8810ac38bfb47e8c49437055b600dfff` and `a8ad324d645d6f7215f207f2cc2988dc49859698` (and preferably `fetchpatch2` / `?full_index=1`, but that’s not really important).

Reply:
Oh, sorry, I forgot to mention: you should use the upstream repo owner in the URLs too, as it’s stabler (forks can get deleted).